src/bridging/BridgingSdk/getQuoteWithBridging.ts

```diff
@@ -0,0 +1,194 @@
+  GetErc20Decimals,
+  QuoteBridgeRequest,
+  QuoteBridgeRequestWithoutAmount,
+} from '../types'
+import { Signer } from '@ethersproject/abstract-signer'
+import { getSigner } from '../../common/utils/wallet'
+
+export async function getQuoteWithBridge<T extends BridgeQuoteResult>(params: {
+  quoteBridgeRequest: QuoteBridgeRequest
+  advancedSettings?: SwapAdvancedSettings
+  provider: BridgeProvider<T>
+  tradingSdk: TradingSdk
+  getErc20Decimals: GetErc20Decimals
+}): Promise<BridgeQuoteAndPost> {
+  const { provider, quoteBridgeRequest, advancedSettings, getErc20Decimals, tradingSdk } = params
+  const { sellTokenChainId, sellTokenAddress, amount, signer: signerLike, ...rest } = quoteBridgeRequest
+  const signer = getSigner(signerLike)
+
+  // Get the mocked hook (for estimating the additional swap costs)
+  const bridgeQuoteRequestWithoutAmount = await getBaseBridgeQuoteRequest({
+    quoteBridgeRequest,
+    provider,
+    getErc20Decimals,
+  })
+
+  // Get the hook mock for cost estimation
+  const gasLimit = provider.getGasLimitEstimationForHook(bridgeQuoteRequestWithoutAmount)
+  const mockedHook = getHookMockForCostEstimation(gasLimit)
+
+  const { sellTokenAddress: intermediateToken, sellTokenDecimals: intermediaryTokenDecimals } =
+    bridgeQuoteRequestWithoutAmount
+
+  // Estimate the expected amount of intermediate tokens received in CoW Protocol's swap
+  const swapParams: WithPartialTraderParams<TradeParameters> = {
+    ...rest,
+    chainId: sellTokenChainId,
+    sellToken: sellTokenAddress,
+    buyToken: intermediateToken,
+    buyTokenDecimals: intermediaryTokenDecimals,
+    amount: amount.toString(),
+    signer,
+  }
+  const { result: swapResult, orderBookApi } = await tradingSdk.getQuoteResults(swapParams, {
+    ...advancedSettings,
+    appData: {
+      metadata: {
+        hooks: mockedHook,
+      },
+    },
+  })
+  const intermediateTokenAmount = swapResult.amountsAndCosts.afterSlippage.buyAmount // Estimated, as it will likely have surplus
+
+  // Get the bridge result
+
+  const { bridgeResult, bridgeHook, appData } = await getBridgeResult({
+    quoteBridgeRequest,
+    swapResult,
+    bridgeQuoteRequestWithoutAmount,
+    provider,
+    intermediateTokenAmount,
+    signer,
+  })
+
+  // Update the receiver and appData (both were mocked before we had the bridge hook)
+  swapResult.tradeParameters.receiver = bridgeHook.recipient
+  const { fullAppData, appDataKeccak256 } = await generateAppDataFromDoc(appData)
+  swapResult.appDataInfo = {
+    fullAppData,
+    appDataKeccak256,
+    doc: appData,
+  }
+
+  // Return the quote results with the post swap order function
+  return {
+    swap: swapResult,
+    bridge: bridgeResult,
+    async postSwapOrderFromQuote() {
+      const quoteResults = {
+        result: {
+          ...swapResult,
+          signer,
+        },
+        orderBookApi,
+      }
+
+      return postSwapOrderFromQuote(quoteResults, { ...advancedSettings, appData })
+    },
```

Comment:
Minor nitpick: I'd prefer to pass an anonymous fn instead as I find it more legible, since the returned fn has the same name as the fn called inside. Feel free to ignore.

```suggestion
    postSwapOrderFromQuote: async () => {
      const quoteResults = {
        result: {
          ...swapResult,
          signer,
        },
        orderBookApi,
      }

      return postSwapOrderFromQuote(quoteResults, { ...advancedSettings, appData })
    },
```

Reply:
I think this is a shorter version ecmascript added at some point, to define function props in objects and classes more nitly. I slightly prefer this approach